woocommerce/woocommerce
# [dev] Monorepo: PSR-4 autoloader for local dev-environment

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Context: p1729782705832279-slack-C03CPM3UXDJ

In the local environment, we inject a custom autoloader as part of the `woo-commerce/monorepo-plugin` package functionality.
This way, autoloader is not getting into production, while loading PSR-4 named classes during development (mapping is loaded from composer manifest).

### Testing instructions

- Under `plugins/woocommerce`: remove `composer` directory and execute `composer install`
- Create a new class under `src`, e.g. `Clazz` (do not trigger autoloading dump)
- Test using it, e,g, adding `var_dump( class_exists( \Automattic\WooCommerce\Clazz::class ) )` to `woocommerce.php`
- Ensure the class is usable

- Ensure zip-files generated from trunk and this branch are identical
